spec/services/step_by_step_filter/options_spec.rb

```diff
@@ -1,70 +1,25 @@
 require "rails_helper"
 
 RSpec.describe StepByStepFilter::Options do
-  it "returns filter options with the default selected" do
-    subject = described_class.statuses
-
-    all_option = {
-      text: "All",
-      data_attributes: {
-        show: "all",
-      },
-      selected: true,
-    }
-
-    expect(subject).to include(all_option)
-
-    StepByStepPage::STATUSES.each do |status|
-      status_option = {
-        text: status.humanize,
-        value: status,
-        data_attributes: {
-          show: status,
-        },
-        selected: false,
-      }
-
-      expect(subject).to include(status_option)
+  describe ".statuses" do
+    it "returns a list of status options with an empty all option" do
+      statuses = described_class.statuses.map { |s| s[:value] }
+      expect(statuses).to match [nil] + StepByStepPage::STATUSES
     end
-  end
-
-  it "returns filter options with the selected status" do
-    subject = described_class.statuses("scheduled")
 
-    all_option = {
-      text: "All",
-      data_attributes: {
-        show: "all",
-      },
-      selected: false,
-    }
-
-    expect(subject).to include(all_option)
-
-    scheduled_option = {
-      text: "Scheduled",
-      value: "scheduled",
-      data_attributes: {
-        show: "scheduled",
-      },
-      selected: true,
-    }
-
-    expect(subject).to include(scheduled_option)
-
-    other_options = StepByStepPage::STATUSES.dup - %w(scheduled)
-
-    other_options.each do |status|
-      status_option = {
-        text: status.humanize,
-        value: status,
-        data_attributes: {
-          show: status,
-        },
-        selected: false,
-      }
+    it "defaults to all option selected" do
+      expect(described_class.statuses)
+        .to include(text: "All",
+                    data_attributes: { show: "all" },
+                    selected: true)
+    end
 
-      expect(subject).to include(status_option)
+    it "returns filter options with the selected status" do
+      expect(described_class.statuses("scheduled"))
+        .to include(text: "Scheduled",
+                    value: "scheduled",
+                    data_attributes: { show: "scheduled" },
+                    selected: true)
```

Comment:
In a similar vein, currently all options might have `selected: true` and would currently pass the test suite. Is it worth validating something like:

```ruby
unselected_options = described_class.statuses.reject(&:selected)
expect(unselected_options.count).to eq(described_class.statuses.count - 1)
```

Reply:
Sure, I can see how that's missing. This all seems to be going progressively over-the-top though given this is essentially a view concern and the logic is rather trivial. The previous tests essentially re-implemented most of the method.

Given we already have this covered by a feature test I'm going to just move this logic into a view and kill the class.

Reply:
Ok I've removed it now.